### PR TITLE
[MRG] FIX label vertical alignment can now be specified

### PR DESCRIPTION
**Pull request by NelleV**
This patch adds a new rcParams allowing to set label vertical alignment. The
sole reason for the existance of this new parameter is to allow user to reset
the style to before 2.0 for testing purposes.

This patch provides a fix to a very pick backward compatibility break. Unfortunately, the
patch that introduces this break also regenerated the test images. This leads to the introduction of a private _test_class stylesheet, corresponding to our current test suite, and a classic stylesheet which is backward compatible.

refs #7905

**TODO**
- [x] Document this new parameter

### CODE REVIEW

File: doc/users/dflt_style_changes.rst [hunk 1]
```diff
@@ -18,6 +18,12 @@ values is a single line of python
 See :ref:`customizing-with-matplotlibrc-files` for details about how to
 persistently and selectively revert many of these changes.
 
+.. note::
```
Comment (tacaswell) [inline doc/users/dflt_style_changes.rst]: I would give this a `'YTick alignment'` section in the `'Tick Label Formatting'` section.   We don't have completely throw our selves under the bus in the first section.  I don't think this needs to be called out in a note, merging it into the paragraph above would be enough.

Comment (NelleV) [inline doc/users/dflt_style_changes.rst]: :+1: on the name and section. Finding a better name was on my todo list.

**Comment by NelleV** [comment]
attn @matplotlib/developers 
This patch introduces a new "private" stylesheet called _classic_test. The "classic" stylesheet can now be 'more' backward compatible without breaking our test suite.

**Comment by anntzer** [comment]
What is the argument for adding yet another private rcparam (which the codebase will have to keepa round for a while) rather than regenerating the test suite images (which is a one shot thing)?

**Comment by NelleV** [comment]
@anntzer It is not private. It is actually used in the classic stylesheet. It is for backward compatibility.
See #refs #7905

**Comment by anntzer** [comment]
Ah, I see, sorry for the noise.
Still, could we just not add the "_classic_test" style and regenerate the test images with the new "ytick.alignment: center" rcparam?

**Comment by NelleV** [comment]
It's around 350 images to regenerate. It's makes the git repository grow massively :(

I'll let @tacaswell give the final word on this.

**Comment by WeatherGod** [comment]
For consistency's sake, we should also have an xtick version of this parameter.

I am disappointed that we seemed to have updated our baseline images, thereby causing this discrepancy between our classic style used by our tests and the classic style used by our users, but I agree that it would be much more harmful to regenerate these images for such a tiny change. We run the risk of a more serious error during the regeneration process.

**Comment by tacaswell** [comment]
I am still in favor of the extra style sheet and :+1: on doing x as well.

**Comment by QuLogic** [comment]
With the pytest migration, it would be simpler if you rebased on and targetted `v2.0.x` directly, then resolving the conflict _upwards_ would be less work. Could you do so and add the 'x' version that @tacaswell requested?

**Comment by NelleV** [comment]
This patch is going to be a mess to apply both on master and v2.0.x
I suspect we may have to create two pull requests.

**Comment by NelleV** [comment]
I am closing this in favor of #8081 : same patch, targetting v2.0.x.

**Pull request closed without merging**